### ISSUE TITLE
Embedding a grain in an iframe a same-origin issue?

### ISSUE THREAD

**Issue by rasos** [sandstorm-io/sandstorm#3131]
We have users, that want to embed e.g. a QuickSurvey into their web page. However, they only see a blank box. Is that a same-origin issue, even if the iFrame is not sandboxed?

**Comment by xet7** [comment]
Sandstorm does not allow embedding in iframe currently, see #https://github.com/sandstorm-io/sandstorm/issues/3068 and https://github.com/sandstorm-io/sandstorm/issues/2906 . This same embedding issue is also at https://github.com/sandforms/sandforms/issues/35

**Comment by ocdtrekkie** [comment]
I suspect there are a lot of reasons this will not work. I think there's a cookie you need to have in your browser to access a grain, if I recall correctly, and the subdomains grains are hosted on are arbitrary and spun up on demand.

Just because you can pop a grain frame out into another tab on your browser doesn't mean you can embed it in a page, unfortunately.

**Comment by xet7** [comment]
Yes, configuring iframes would need a lot more settings for controlling many headers, cors, etc and very easily Chrome blocks any kind of suspicious behaviour.

**Comment by xet7** [comment]
It's much easier to add more layout features, additional html css etc to QuickSurvey and just link to that quicksurvey page, than try to get iframes working.
